sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -315,9 +315,21 @@ object GpuDeviceManager extends Logging {
       }
 
       Cuda.setDevice(gpuId)
-      Rmm.initialize(init, logConf, poolAllocation)
-      RapidsBufferCatalog.init(conf)
+      try {
+        Rmm.initialize(init, logConf, poolAllocation)
+      } catch {
+        case ex: Throwable =>
+          if ((init | RmmAllocationMode.CUDA_ASYNC) != 0) {
+            logWarning(
+              "RMM failed to initialize with ASYNC allocator and will fallback to ARENA allocator")
+            init = init & (~RmmAllocationMode.CUDA_ASYNC) & RmmAllocationMode.ARENA
+            Rmm.initialize(init, logConf, poolAllocation)
```

Comment:
Could we wrap this second call to `initialize` in a try catch, and in case of exception, add `ex` (the exception you caught for ASYNC) as a suppressed exception, before you throw the exception seen during ARENA initialization.

Reply:
I would make this last one catch `Throwable` btw, and rethrow it.